src/ProjectOrigin.RequestProcessor/Services/SynchronousInMemoryRequestProcessor.cs

```diff
@@ -0,0 +1,75 @@
+using System.Collections.Concurrent;
+using ProjectOrigin.RequestProcessor.Interfaces;
+using ProjectOrigin.RequestProcessor.Models;
+using ProjectOrigin.VerifiableEventStore.Models;
+using ProjectOrigin.VerifiableEventStore.Services.Batcher;
+
+namespace ProjectOrigin.RequestProcessor.Services;
+
+public class SynchronousInMemoryRequestProcessor : IRequestProcessor
+{
+    private ConcurrentDictionary<RequestId, RequestResult> requestDictionary = new();
+    private IDispatcher dispatcher;
+    private IBatcher batcher;
+    private IEventSerializer eventSerializer;
+
+    public SynchronousInMemoryRequestProcessor(IDispatcher dispatcher, IBatcher batcher, IEventSerializer eventSerializer)
+    {
+        this.dispatcher = dispatcher;
+        this.batcher = batcher;
+        this.eventSerializer = eventSerializer;
+    }
+
+    public async Task QueueRequest(IPublishRequest request)
+    {
+        if (requestDictionary.TryAdd(request.RequestId, new RequestResult(request.RequestId, RequestState.Queued)))
+        {
+            await ProcessRequest(request);
+        }
+        else
+        {
+            throw new ArgumentException($"Request already exists");
+        }
+    }
+
+    public Task<RequestResult> GetRequestStatus(RequestId id)
+    {
+        if (requestDictionary.TryGetValue(id, out var requestStatus))
+        {
+            return Task.FromResult(requestStatus);
+        }
+        else
+        {
+            return Task.FromResult(new RequestResult(id, RequestState.Unknown));
+        }
+    }
+
+    private async Task ProcessRequest(IPublishRequest request)
+    {
+        var (result, nextEventIndex) = await dispatcher.Verify(request);
+
+        if (result.IsValid)
+        {
+            SetState(request.RequestId, RequestState.Processing);
+            var serializedEvent = eventSerializer.Serialize(new EventId(request.FederatedStreamId.StreamId, nextEventIndex), request.Event);
+
+            await batcher.PublishEvent(serializedEvent);
+            SetState(request.RequestId, RequestState.Completed);
+        }
+        else
+        {
+            if (result.ErrorMessage is null)
+                throw new InvalidOperationException("Verification failed without errorMessage!");
+
+            SetState(request.RequestId, RequestState.Failed, result.ErrorMessage);
+        }
+    }
+
+    private void SetState(RequestId id, RequestState state, string? exception = null)
```

Comment:
? [Ternary conditional operator](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/conditional-operator) - validates boolean expression

Reply:
It has more functions, in this case it tells that the string is nullable, otherwise the value must hold a value.

Reply:
in this case a null value exception is returned if the string is null - or am I wrong